Kernel/Thread.cpp

```diff
@@ -927,11 +926,12 @@ void Thread::set_state(State new_state, u8 stop_signal)
         auto& process = this->process();
         if (process.set_stopped(true) == false) {
             process.for_each_thread([&](auto& thread) {
-                if (&thread == this || thread.is_stopped())
-                    return IterationDecision::Continue;
+                if (&thread == this)
+                    return;
+                if (!thread.is_stopped())
+                    return;
```

Comment:
This inverts the `thread.is_stopped()` check!

Reply:
Thank you for pointing this out. Fixed in the latest commit.